setup.sh

```diff
@@ -0,0 +1,617 @@
+  writeBold "$RED" "    to be the available amount on your system (${CHECK_MEM})."
+  write            "    Sorry, you cannot launch Kuzzle on this machine."
+  echo
+  exit 5
+fi
+
+write "$GREEN" "[✔] Available memory is at least 4Gb."
+
+CHECK_CORES=$(awk '/^processor/{print $3}' /proc/cpuinfo | tail -1)
+MIN_CORES=4
+if (( "$CHECK_CORES" < "$MIN_CORES" )); then
+ echo
+  writeBold "$RED" "[✖] Kuzzle needs at least $MIN_CORES processor cores, which does not seem"
+  writeBold "$RED" "    to be the available amount on your system."
+  write            "    Sorry, you cannot launch Kuzzle on this machine."
+  echo
+  exit 6
+fi
+
+write "$GREEN" "[✔] At least 4 processor cores available."
+
+if [ "$EUID" -ne 0 ]; then
+  echo
+  writeBold "$YELLOW" "[✖] This script needs to be executed with root privileges."
+  write               "    Try typing: sudo $0"
+  echo
+  exit 1
+fi
+
+write "$GREEN" "[✔] Script has root privileges."
+
+if ! commandExists curl && ! commandExists wget; then
+  INSTALL_DL=1
+  write "$YELLOW" "[✖] cUrl is not installed."
+  exit 1
+elif commandExists curl; then
+  setDL "curl"
+  write "$GREEN" "[✔] cUrl is installed."
+elif commandExists wget; then
+  setDL "wget"
+  write "$GREEN" "[✔] Wget is installed."
+fi
+
+if ! commandExists docker; then
+  write "$YELLOW" "[✖] Docker is not installed."
+  INSTALL_DOCKER=1
+elif [[ $? == 2 ]]; then
+  MIN_DOCKER_VER=1.12.0
+  dockerVersion=$(docker -v | perl -nle 'm/(\d+(\.\d*(\.\d*)?)?)/;print $1')
+  vercomp ${dockerVersion} $MIN_DOCKER_VER
+  write "$YELLOW" "[✖] The current version of Docker ${dockerVersion} is older"
+  write "$YELLOW"  "    than the required one ($MIN_DOCKER_VER+)."
+  INSTALL_DOCKER=1
+elif ! $INSTALL_DOCKER && ! $(docker run hello-world &> /dev/null); then
+  echo
+  writeBold "$RED" "[✖] Docker does not seem to be running on your system."
+  writeBold "$RED" "    Please start the Docker daemon and re-run this script"
+  write            "    More information at https://docs.docker.com/engine/admin/"
+  echo
+  exit 5
+else
+  write "$GREEN" "[✔] Docker is installed and running."
+fi
+
+if ! commandExists docker-compose; then
+  write "$YELLOW" "[✖] Docker Compose is not installed."
+  INSTALL_DOCKER_COMPOSE=1
+elif [[ $? == 2 ]]; then
+  MIN_DOCKER_COMPOSE_VER=1.8.0
+  dockerComposeVersion=$(docker-compose -v | perl -nle 'm/(\d+(\.\d*(\.\d*)?)?)/;print $1')
+  vercomp ${dockerComposeVersion} $MIN_DOCKER_COMPOSE_VER
+  write "$YELLOW" "[✖] The current version of Docker ${dockerComposeVersion} is older"
+  write "$YELLOW"  "    than the required one ($MIN_DOCKER_COMPOSE_VER+)."
+  INSTALL_DOCKER_COMPOSE=1
+else
+  write "$GREEN" "[✔] Docker Compose is installed."
+fi
+
+CHECK_MAP_COUNT=$(sysctl -a 2> /dev/null | grep vm.max_map_count | cut -d'=' -f2 | tr -d ' ')
+REQUIRED_MAP_COUNT=262144
+if [ -z "${CHECK_MAP_COUNT}" ] || [ ${CHECK_MAP_COUNT} -lt $REQUIRED_MAP_COUNT ]; then
```

Comment:
Woops, nope. `vm.max_map_count` is not allowed in docker containers due to shared kernel namespace